File: 122/py/122.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2 :
            return 0

        buy = prices[0]
        profit = 0

        for price in prices :
            if price > buy:
                profit += price - buy
            buy = price

        return profit

File: 122/go/122.go
func maxProfit(prices []int) int {
	
	buy := prices[0]
	profit := 0

	if len(prices) < 2 {
		return profit
	}
	
	for _, price := range prices {
		if price > buy {
			profit += price - buy
		}
		buy = price
	}
	return profit
}